File: examples/go/add-raw-output/RawOutputTest.runar.go
package contract

import runar "github.com/icellan/runar/packages/runar-go"

// RawOutputTest exercises AddRawOutput alongside AddOutput for stateful
// contracts.
type RawOutputTest struct {
	runar.StatefulSmartContract
	Count runar.Bigint
}

// SendToScript emits a raw output with arbitrary script bytes, then increments
// the counter and emits the state continuation.
func (c *RawOutputTest) SendToScript(scriptBytes runar.ByteString) {
	c.AddRawOutput(1000, scriptBytes)
	c.Count = c.Count + 1
	c.AddOutput(0, c.Count)
}

File: examples/rust/add-raw-output/RawOutputTest.runar.rs
use runar::prelude::*;

/// RawOutputTest -- Exercises add_raw_output alongside add_output for stateful
/// contracts.
#[runar::contract]
pub struct RawOutputTest {
    pub count: Bigint,
}

#[runar::methods(RawOutputTest)]
impl RawOutputTest {
    /// Emit a raw output with arbitrary script bytes, then increment the
    /// counter and emit the state continuation.
    #[public]
    pub fn send_to_script(&mut self, script_bytes: ByteString) {
        self.add_raw_output(1000, script_bytes);
        self.count = self.count + 1;
        self.add_output(0, self.count);
    }
}
